### PR TITLE
Update Spring RestTemplate docs.

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
Reflect changes from https://github.com/getsentry/sentry-java/pull/1246 and https://github.com/getsentry/sentry-java/issues/1240

### CODE REVIEW

**Comment by maciejwalkowiak** [comment]
cc @marandaneto

**Review by marandaneto — APPROVED** [review]
LGTM, let's merge only after released

**Comment by marandaneto** [comment]
@maciejwalkowiak quick question, people that do have a `UriTemplateHandler` already set in the previous version, would need to remove it or it'd work ootb? if so, we'd need to add it to the migration page.

**Comment by marandaneto** [comment]
> @maciejwalkowiak quick question, people that do have a `UriTemplateHandler` already set in the previous version, would need to remove it or it'd work ootb? if so, we'd need to add it to the migration page.

^

**Comment by maciejwalkowiak** [comment]
Yes, they need to remove it. I guess we won't keep anymore migration page as a migration from 1.x to latest, but create sections for each version?

**Comment by marandaneto** [comment]
> Yes, they need to remove it. I guess we won't keep anymore migration page as a migration from 1.x to latest, but create sections for each version?

not sure the best approach either, but we need to figure this out otherwise people won't know what to do, I believe a section for a minor version is a good start.

File: src/platforms/java/migration.mdx [hunk 1]
```diff
@@ -4,6 +4,38 @@ sidebar_order: 1000
 description: "Learn about migrating from io.sentry:sentry 1.x to io.sentry:sentry 3.x."
 ---
 
+## Migrating from `io.sentry:sentry` `4.1.0` to `io.sentry:sentry` `4.1.1`
```
Comment (marandaneto) [inline src/platforms/java/migration.mdx:7]:
```suggestion
## Migrating from `io.sentry:sentry` `4.1.0` to `io.sentry:sentry` `4.2.0`
```

let's bump a minor, a few breaking changes

File: src/platforms/java/migration.mdx [hunk 2]
```diff
@@ -4,6 +4,82 @@ sidebar_order: 1000
 description: "Learn about migrating from io.sentry:sentry 1.x to io.sentry:sentry 3.x."
 ---
 
+## Migrating from `io.sentry:sentry` `4.1.0` to `io.sentry:sentry` `4.2.0`
+
+`operation` is now a required property of the `SentryTransaction` and `SentrySpan`. Whenever a transaction or span is started, the value for `operation` must be provided:
+
+```java
+Sentry.startTransaction("transaction-name", "operation-name");
+```
```
Comment (marandaneto) [inline src/platforms/java/migration.mdx:13]: this should be also under https://github.com/getsentry/sentry-docs/blob/master/src/platforms/android/migration.mdx

**Review by marandaneto — APPROVED** [review]

**Pull request merged**